pallets/bridge-transfer/src/lib.rs

```diff
@@ -93,27 +97,20 @@ pub mod pallet {
 			amount: BalanceOf<T>,
 			rid: ResourceId,
 		) -> DispatchResult {
-			let source = T::BridgeOrigin::ensure_origin(origin)?;
-			// transfer to bridge account from external accounts is not allowed.
-			if source == to {
-				fail!(Error::<T>::InvalidCommand);
-			}
+			T::BridgeOrigin::ensure_origin(origin)?;
 
+			let total_issuance = <T as Config>::Currency::total_issuance();
+			if total_issuance + amount > T::MaximumIssuance::get() {
```

Comment:
do we need to check overflow here? Like `checked_add`? For example the `issue()` fn in pallet-balances: https://github.com/paritytech/substrate/blob/master/frame/balances/src/lib.rs#L1453

Or can we remove this check and rely on the error handling of `mint_into()`?

Reply:
agree, overflow check is needed.